### DISCUSSION TITLE
Mapping bindings to different visual modes

### DISCUSSION THREAD

**Discussion by givensuman** [LazyVim/LazyVim#4215] [Q&A] [answered]
I'd like to map a binding to different visual modes (line, block, etc.) Presently this is my indentation keybinding:

```lua
local function map(mode, lhs, rhs, opts)
  local options = { noremap = true, silent = true }
  if opts then
    if opts.desc then
      opts.desc = "keymaps.lua: " .. opts.desc
    end
    options = vim.tbl_extend("force", options, opts)
  end
  vim.keymap.set(mode, lhs, rhs, options)
end

map("v", "<TAB>", ">gv")
map("v", "<S-TAB>", "<gv")
```
This works, but not very well in line and block mode. Running `:help mode()` indicates those modes are represented by `V` and `CTRL-V`, respectively. However,

```lua
map("V", "<TAB>", ">gv")
map("V", "<S-TAB>", "<gv")
map("CTRL-V", "<TAB>", ">gv")
map("CTRL-V", "<S-TAB>", "<gv")
```
throws an error from LazyVim, telling me these aren't valid shortnames. What should I be doing in this situation?

**Comment by givensuman** [comment]
If this isn't possible, 

```lua
vim.cmd("vnoremap <TAB> >gv")
vim.cmd("vnoremap <S-TAB> <gv")
```
works for now

**Comment by dpetka2001** [comment]
Just remove `noremap = true` from your `options`. `vim.keymap.set` doesn't take a `noremap` option, but a `remap` option which by default is set to `false`. `noremap` is an option for `nvim_set_keymap`.

**Reply by givensuman** [reply]
If this is what you meant

```lua
local function map(mode, lhs, rhs, opts)
  local options = { silent = true }
  if opts then
    if opts.desc then
      opts.desc = "keymaps.lua: " .. opts.desc
    end
    options = vim.tbl_extend("force", options, opts)
  end
  vim.keymap.set(mode, lhs, rhs, options)
end

map("v", "<TAB>", ">gv")
map("v", "<S-TAB>", "<gv")
-- :help mode()
map("V", "<TAB>", ">gv")
map("V", "<S-TAB>", "<gv")
map("CTRL-V", "<TAB>", ">gv")
map("CTRL-V", "<S-TAB>", "<gv")
```
still produces the same error

**Reply by dpetka2001** [reply, marked as answer]
You can't use `V` and `<ctrl-v>`. Please read `:h map-modes`.
